components/Visualization/index.tsx

```diff
@@ -1,17 +1,11 @@
-import { removeHyphens } from '@/utils/strings/removeHyphens';
-import { useRouter } from 'next/router';
 import React from 'react'
 import StepsSlider from '../StepsSlider';
-import visualizationFactory from './visualizationFactory';
 
-export const Visualization = () => {
-    const router = useRouter();
-    const { dataStructure } = router.query;
-    const visualizationSteps = visualizationFactory(dataStructure)
+export const Visualization = ({visualization}) => {
     return (
         <>
-            <h3 className="text-capitalize mt-4">{`${removeHyphens(dataStructure)} - visualization`}</h3>
-            {visualizationSteps && <StepsSlider steps={visualizationSteps} />}
+            
+            {visualization && <StepsSlider visualization={visualization} />}
```

Comment:
here we also can rename this 'visualization' to something more descriptive. I know I named it this way, but I was writing that in a rush